handlers/handlers.go

```diff
@@ -256,17 +256,36 @@ type AuthentictedRequestHandlerType func(
 	*http.Request,
 	models.UserId)
 
-func AuthenticateOrRedirectToLogin(
+func AuthenticateOrRedirect(
 	authenticatedHandlerFunc AuthentictedRequestHandlerType,
+	redirectPath string,
 ) http.HandlerFunc {
 	return func(responseWriter http.ResponseWriter, request *http.Request) {
 		if userId, err := getUserIdFromJwtToken(request); err != nil {
+			switch request.Method {
 			// If not logged in, redirect to login page
-			http.Redirect(
-				responseWriter,
-				request,
-				paths.LoginOrSignupPage,
-				http.StatusTemporaryRedirect)
+			case http.MethodGet:
+				http.Redirect(
+					responseWriter,
+					request,
+					redirectPath,
+					http.StatusTemporaryRedirect)
+				return
+			default:
+				respondWithMethodNotAllowed(responseWriter, http.MethodGet)
+			}
+		} else {
+			authenticatedHandlerFunc(responseWriter, request, userId)
+		}
+	}
+}
+
+func AuthenticateOrUnauthorized(
```

Comment:
how about `AuthenticateOrReturnUnauthorized` a bit verbose but clear no?

Reply:
I don't particularly like having the verb return in a function, but given that i have no alternatives and the function name isn't wrong let's go with it for now

Reply:
I think I just realized what pattern I'd like to end up at eventually. Typically an issue with naming a function is an issue the function was poorly designed. This function is doing two things. Ideally it would just authenticate, and return an error if it failed to do so. We did this as a convenience to not have to put in error handling that returns 401.

I totally like the idea, but I think what we want to do eventually is have some wrapper around all of our handlers that just checks the error that was returned and translates that to an http response code. For example, this authenticate function would return an error like `FailedToAuthenicateUserError` or `UserIsUnauthenticatedError`, and another function might return something like `MalformedPostBodyContentError`. And then the wrapper function would look at the error type and say okay, you're a 401, you're a 400, etc.